application/src/main/kotlin/net/corda/v5/application/crypto/DigitalSigningService.kt

```diff
@@ -0,0 +1,53 @@
+package net.corda.v5.application.crypto
+
+import net.corda.v5.base.annotations.DoNotImplement
+import net.corda.v5.base.annotations.Suspendable
+import net.corda.v5.crypto.*
+import java.security.KeyPair
+import java.security.PrivateKey
+import java.security.PublicKey
+import java.util.*
+
+/**
+ * Responsible for storing and using private keys to sign things. An implementation of this may, for example, call out
+ * to a hardware security module that enforces various auditing and frequency-of-use requirements.
+ *
+ * Corda provides an instance of [DigitalSignatureVerificationService] to flows via property injection.
+ */
+@DoNotImplement
+interface DigitalSigningService {
```

Comment:
Maybe its just me, but `DigitalSigningService` sounds a bit odd to me. I'd prefer `DigitalSingatureService` (rolls off the tongue better) but then that sounds like a more general service and might confuse things since we have `DigitalSignatureVerificationService`. Should we go as far as to name it `DigitalSingatureSigningService`? That would pair it up with the verification version more closely.